eventmetrics/parsers/provide.go

```diff
@@ -76,58 +105,144 @@ func provideParsers() fx.Option {
 			},
 		},
 		fx.Annotated{
-			Group:  "parsers,flatten",
-			Target: TimeElapsedParsers,
+			Group: "parsers",
+			Target: func(cycleValidator history.CycleValidator, eventValidator validation.Validator, parserIn RebootParserIn) queue.Parser {
+				name := "reboot_duration_parser"
+				logger := parserIn.Logger.With(zap.String("parser", name))
+				comparators := history.Comparators([]history.Comparator{
+					history.OlderBootTimeComparator(),
+				})
+
+				rebootEventFinder := history.LastSessionFinder(validation.DestinationValidator(rebootPendingEventType))
+				parserValidators := createParserValidators(cycleValidator, eventValidator, logger, parserIn.Measures)
+				return &RebootDurationParser{
+					name:                 name,
+					relevantEventsParser: history.LastCycleToCurrentParser(comparators),
+					parserValidators:     parserValidators,
+					calculators: []DurationCalculator{
+						BootDurationCalculator(logger, createBootDurationCallback(parserIn.Measures)),
+						&EventToCurrentCalculator{
+							logger:          logger,
+							successCallback: createRebootToManageableCallback(parserIn.Measures),
+							eventFinder:     rebootEventFinder,
+						},
+					},
+					measures: parserIn.Measures,
+					client:   parserIn.CodexClient,
+					logger:   logger,
+				}
+			},
 		},
 	)
 }
 
-// TimeElapsedParsers creates a list of TimeElapsedParsers from the config.
-func TimeElapsedParsers(parsers TimeElapsedParsersIn) ([]queue.Parser, error) {
-	if valid, err := validNames(parsers.Config.Parsers); !valid {
-		return nil, err
+func createBootDurationCallback(m Measures) func(interpreter.Event, float64) {
```

Comment:
Technically the measure's fields could be nil...I'm not sure if we need to check them since I'm fairly certain we've never checked them anywhere else (usually we use `*Measures` and only nil check that).